kinoshita/pracitce-erd/instagram/laravel/app/Models/Comment.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Comment extends Model
+{
+    use HasFactory;
+
+    public function user()
+    {
+        return $this->belongsTo(User::class);
+    }
+
+    public function commentable()
+    {
+        return $this->morphTo();
+    }
+
+    public function post()
+    {
+        return $this->morphOne(Post::class, 'commentable'); 
+    }
+
+    public function comments() // コメントに対する返信コメント
```

Comment:
リレーション名はカスタムで設定できますよ！このままだと、commentableとcommentsが見分けしにくくなりますね。今回みたいに同じモデルを指すリレーションなら尚更見分けやすいですよ。

「返信」なので、`replies`にしましょう

```suggestion
    public function replies()
```

Reply:
こちらで修正しました

[137a5d1](https://github.com/StoryEdutech/study_repo/pull/88/commits/137a5d10b8b65bd78e8118f709b8001819259c27)